search/searcher/geoshape_contains_test.go

```diff
@@ -0,0 +1,938 @@
+		Desc             string
+		QueryType        string
+	}{
+		{
+			QueryShape:       [][][]float64{{{0.5, 0.5}, {1, 0.5}, {1, 1}, {0.5, 1}, {0.5, 0.5}}},
+			DocShapeVertices: [][]float64{{0, 1}, {1, 0}},
+			DocShapeName:     "envelope1",
+			Expected:         nil,
+			Desc:             "polygon contained inside envelope with edge overlaps", // this fails since
+			// contains doesn't include edges or vertices
+			QueryType: "contains",
+		},
+		{
+			QueryShape:       [][][]float64{{{0.25, 0.25}, {0.5, 0.25}, {0.5, 0.5}, {0.25, 0.25}, {0.25, 0.25}}},
+			DocShapeVertices: [][]float64{{0, 1}, {1, 0}},
+			DocShapeName:     "envelope1",
+			Expected:         []string{"envelope1"},
+			Desc:             "polygon contained completely inside envelope",
+			QueryType:        "contains",
+		},
+	}
+
+	i := setupIndex(t)
+
+	for _, test := range tests {
+		indexReader, closeFn, err := testCaseSetup(t, test.DocShapeName, "envelope",
+			[][][][]float64{{test.DocShapeVertices}}, i)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		t.Run(test.Desc, func(t *testing.T) {
+			got, err := runGeoShapePolygonQueryWithRelation(test.QueryType,
+				indexReader, test.QueryShape, "geometry")
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !reflect.DeepEqual(got, test.Expected) {
+				t.Errorf("expected %v, got %v for polygon: %+v",
+					test.Expected, got, test.QueryShape)
+			}
+		})
+
+		err = closeFn()
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+	}
+}
+
+func TestMultiPointPolygonContains(t *testing.T) {
+	tests := []struct {
+		QueryShape       [][]float64
+		DocShapeVertices [][][]float64
+		DocShapeName     string
+		Expected         []string
+		Desc             string
+		QueryType        string
+	}{
+		{
+			QueryShape:       leftRectEdgeMultiPoint,
+			DocShapeVertices: leftRectWithHole,
+			DocShapeName:     "polygon1",
+			Expected:         []string{"polygon1"},
+			Desc:             "multi point inside polygon with hole",
+			QueryType:        "contains",
+		},
+		{
+			QueryShape:       [][]float64{{1, 0.5}},
+			DocShapeVertices: rightRect,
+			DocShapeName:     "polygon1",
+			Expected:         nil,
+			Desc:             "multi point on polygon edge",
+			QueryType:        "contains",
+		},
+	}
+
+	i := setupIndex(t)
+
+	for _, test := range tests {
+		indexReader, closeFn, err := testCaseSetup(t, test.DocShapeName, "polygon",
+			[][][][]float64{test.DocShapeVertices}, i)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		t.Run(test.Desc, func(t *testing.T) {
+			got, err := runGeoShapePointRelationQuery(test.QueryType,
+				true, indexReader, test.QueryShape, "geometry")
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !reflect.DeepEqual(got, test.Expected) {
+				t.Errorf("expected %v, got %v for multipoint: %+v",
+					test.Expected, got, test.QueryShape)
+			}
+		})
+
+		err = closeFn()
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+	}
+}
+
+func TestMultiPointLinestringContains(t *testing.T) {
```

Comment:
Since the XXXX_LinestringContains mostly serves as a reference returning nothing, can all these variations could be made to a single test TestLinestringContains() with all entries in a single table?

Reply:
I'm not sure it's possible to combine those functions since the querying functions and fields change based on query shape.

Reply:
I'm not sure these can be combined since the query shapes and query functions are different for each.